cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/sql/PostgresSqlStructuredTable.java

```diff
@@ -458,6 +458,28 @@ public void deleteAll(Range keyRange) throws InvalidFieldException, IOException
     }
   }
 
+  @Override
+  public long count(Range keyRange) throws IOException {
+    LOG.trace("Table {}: count with range {}", tableSchema.getTableId(), keyRange);
+    String sql = getCountStatement(keyRange);
+    try (PreparedStatement statement = connection.prepareStatement(sql)) {
+      setStatementFieldByRange(keyRange, statement);
+      LOG.trace("SQL statement: {}", statement);
+
+      statement.executeQuery();
+      try (ResultSet resultSet = statement.executeQuery()) {
+        if (!resultSet.next()) {
+          throw new IOException(String.format("Failed to get count from table %s",
```

Comment:
Why it is an exception? Shouldn't it be 0?

Reply:
returned 0, logged warning.